wasm/jit/jit_arm64_test.go

```diff
@@ -2439,3 +2449,164 @@ func TestArm64Compiler_compileLoad(t *testing.T) {
+}
+
+func TestAmd64Compiler_compileMaybeGrowValueStack(t *testing.T) {
+	t.Run("not grow", func(t *testing.T) {
+		const stackPointerCeil = 5
+		for _, baseOffset := range []uint64{5, 10, 20} {
+			t.Run(fmt.Sprintf("%d", baseOffset), func(t *testing.T) {
+				env := newJITEnvironment()
+				compiler := env.requireNewCompiler(t)
+
+				// The assembler skips the first instruction so we intentionally add NOP here.
+				// TODO: delete after #233
+				compiler.compileNOP()
+
+				err := compiler.compileMaybeGrowValueStack()
+				require.NoError(t, err)
+				require.NotNil(t, compiler.onStackPointerCeilDeterminedCallBack)
+
+				valueStackLen := uint64(len(env.stack()))
+				stackBasePointer := valueStackLen - baseOffset // Ceil <= valueStackLen - stackBasePointer = no need to grow!
+				compiler.onStackPointerCeilDeterminedCallBack(stackPointerCeil)
+				compiler.onStackPointerCeilDeterminedCallBack = nil
+				env.setValueStackBasePointer(stackBasePointer)
+
+				compiler.compileExitFromNativeCode(jitCallStatusCodeReturned)
+
+				// Generate and run the code under test.
+				code, _, _, err := compiler.compile()
+				require.NoError(t, err)
+				env.exec(code)
+
+				// The status code must be "Returned", not "BuiltinFunctionCall".
+				require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
+			})
+		}
+	})
+	t.Run("grow", func(t *testing.T) {
+		env := newJITEnvironment()
+		compiler := env.requireNewCompiler(t)
+
+		// The assembler skips the first instruction so we intentionally add NOP here.
+		// TODO: delete after #233
+		compiler.compileNOP()
+
+		err := compiler.compileMaybeGrowValueStack()
+		require.NoError(t, err)
+
+		// On the return from grow value stack, we simply return.
+		compiler.compileReturnFunction()
+
+		stackPointerCeil := uint64(6)
+		compiler.stackPointerCeil = stackPointerCeil
+		valueStackLen := uint64(len(env.stack()))
+		stackBasePointer := valueStackLen - 5 // Ceil > valueStackLen - stackBasePointer = need to grow!
+		env.setValueStackBasePointer(stackBasePointer)
+
+		// Generate and run the code under test.
+		code, _, _, err := compiler.compile()
+		require.NoError(t, err)
+		env.exec(code)
+
+		// Check if the call exits with builtin function call status.
+		require.Equal(t, jitCallStatusCodeCallBuiltInFunction, env.jitStatus())
+
+		// Reenter from the return address.
+		returnAddress := env.callFrameStackPeek().returnAddress
+		require.NotZero(t, returnAddress)
+		jitcall(returnAddress, uintptr(unsafe.Pointer(env.engine())))
+
+		// Check the result. This should be "Returned".
+		require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
+	})
+}
+
+func TestArm64Compiler_compieleHostFunction(t *testing.T) {
+	env := newJITEnvironment()
+	compiler := env.requireNewCompiler(t)
+
+	// The assembler skips the first instruction so we intentionally add NOP here.
+	// TODO: delete after #233
+	compiler.compileNOP()
+
+	addr := wasm.FunctionAddress(100)
```

Comment:
is it too much to actually call a go function?

Reply:
too much and doesn't affect anything in the test, so I will leave as-is